syncano_cli/parse_to_syncano/migrations/relation.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# a relation helper
+from parse_to_syncano.config import PARSE_PAGINATION_LIMIT
+from parse_to_syncano.migrations.aggregation import data_aggregate
+from parse_to_syncano.migrations.mixins import PaginationMixin, ParseConnectionMixin
+from parse_to_syncano.processors.klass import ClassProcessor
+from syncano.models import Object
+from syncano_cli import LOG
+
+
+class RelationProcessor(ParseConnectionMixin, PaginationMixin):
+
+    def __init__(self, relations, *args, **kwargs):
+        super(RelationProcessor, self).__init__(*args, **kwargs)
+        self.relations = relations
+        self.reference_map = data_aggregate.reference_map
+
+    def process(self, instance):
+        LOG.info('Processing relations...')
+        for class_relation in self.relations:
+            for class_name, relations in class_relation.iteritems():
+                for relation in relations:
+                    for field_name, relation_meta in relation.iteritems():
+                        target_name = relation_meta['targetClass']
+
+                        # get the parse classes now;
+                        for parse_class_name, objects_id_map in self.reference_map.iteritems():
+                            if class_name == ClassProcessor.normalize_class_name(parse_class_name):
+                                for parse_id, syncano_id in objects_id_map.iteritems():
+                                    limit, skip = self.get_limit_and_skip()
+
+                                    while True:
+
+                                        objects = self.get_parse_objects(parse_class_name, parse_id, field_name,
```

Comment:
Wohhooo :) 8-th level (from method definition), one more and you would hit the bottom of hell. 
Split it into methods :)